### ISSUE TITLE
Intl: resolvedOptions().locale does not correctly represent the collation value

### ISSUE THREAD

**Issue by jackhorton** [Microsoft/ChakraCore#4031]
```js
new Intl.Collator("de-DE-u-co-phonebk").resolvedOptions()
```

Expected result (as in Chrome and Firefox)
```js
{
    locale: "de-u-co-phonebk",
    collation: "phonebk"
    ...
}
```

Actual result (Edge and IE11)
```js
{
    locale: "de-DE",
    collation: "phonebk"
    ...
}
```

This is caused by some very special/intricate/poor handling of locales that we do for collation options, since the system API that we use on Windows for comparing strings based on locale expects the language tag in a different format than the Intl spec works in. @bterlson is the locale value that we report here actually incorrect?

**Comment by dilijev** [comment]
IMO it's probably not "incorrect" per se, because the spec at present is pretty much all "best effort" -- as long as the behavior is correct (which again by spec is best-effort but in practice for us means if we support that collator, we should use that collator as requested).

In terms of conformance, other engines are still reporting the collator extension (in this case `-u-co-phonebk`) if it is supported, so we should probably do this too.

However, also IMO, it is misleading to report the resolved locale in a way that shows we do not support the collation option that we did in fact select to use (region part of the locale (`-DE`) aside).

**Comment by jackhorton** [comment]
For what its worth, we do this for all supported unicode tags in the collator:

```js
new Intl.Collator("de-u-co-phonebk-kn-true-kf-upper").resolvedOptions()
```

Firefox and Chrome return `{ locale: "de-u-co-phonebk-kn-true-kf-upper", ... }` while Chakra returns `{ locale: "de" ... }`

**Comment by jackhorton** [comment]
Thats a lie -- Firefox returns that, chrome will only ever keep -co- on the locale string, and they will always strip off everything else.

**Comment by dilijev** [comment]
@jackhorton Chrome's behavior makes more sense to me, since the value of the options object input to the ctor overrides. Since those are also on the resolved options, removing ambiguity by de-duping them (removing from the locale string) seems to make sense.

**Comment by dilijev** [comment]
@jackhorton is this fixed?

**Comment by jackhorton** [comment]
Closed in #4083, but this is being improved in an upcoming pull that standardizes everything around the spec version of ResolveLocale
